common/utils.py

```diff
@@ -208,12 +208,22 @@ def inspect_directory(files: List[str], pattern: Optional[str] = None,
         else:
             continue
 
-def compile_code():
+def compile_code(makefile_target: str = ""):
     """Compiles the current directory (either with Make or manually)"""
-    ls_output = os.listdir()
-    if "Makefile" not in ls_output:
-        # Let's let the grader figure it out
-        os.system("bash")
+    if makefile_target == "":
+        # When no makefile target is provided then
+        # default to original behavior of checking current folder
+        ls_output = os.listdir()
+        if "Makefile" not in ls_output:
+            # Let's let the grader figure it out
+            os.system("bash")
+    else:
+        # When makefile target is provided then
+        # check provided folder for makefile
+        ls_output = os.listdir(makefile_target)
+        if "Makefile" not in ls_output:
+            # Let's let the grader figure it out
+            os.system("bash")
 
     p.print_cyan("[ Compiling... ]")
     ret = subprocess.call(MAKE, shell=True)
```

Comment:
I believe what we want here is something like

`MAKE.format(makefile_target)`. We'll have to change the `MAKE` global to have a `{}` in it.

This will also take care of the case where `makefile_target` is "".

Reply:
Makefiles are new to me but this makes a lot more sense than what I was trying to do.